module/VuFind/src/VuFind/Content/Covers/Bokinfo.php

```diff
@@ -0,0 +1,200 @@
+    }
+
+    /**
+     * Get image URL for a particular API key and set of IDs (or false if invalid).
+     *
+     * @param string $key  API key
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        if (!isset($ids['isbn'])) {
+            return false;
+        }
+        if (!isset($key)) {
+            return false;
+        }
+
+        $isbn = $ids['isbn']->get13();
+        try {
+            $client = $this->createHttpClient(
+                "https://api.bokinfo.se/book/get/$isbn"
+            );
+
+            $client->getRequest()->getHeaders()
+                  ->addHeaderLine("Ocp-Apim-Subscription-Key", $key);
+
+            try {
+                $resp= $client->send();
+                $body = $resp->getBody();
+                $url = $this->getImageUrl($body);
+            } catch (\Exception $e) {
+            }
+
+            if ($this->testUrlFunction($url)) {
+                return "$url";
+            }
+        } catch (\Throwable $ex) {
+            return false;
+        } catch (\Exception $ex) {
+            return false;
+        } catch (\RuntimeException $ex) {
+            return false;
+        }
+
+        return false;
+    }
+
+    /**
+     * Return a HTTP Client object
+     *
+     * @param string $url API Url
+     *
+     * @return HttpClient Http Client
+     */
+    protected function createHttpClient($url)
+    {
+        $client = $this->httpService->createClient($url);
+
+        $adapter = $client->getAdapter();
+        if ($adapter instanceof \Laminas\Http\Client\Adapter\Socket) {
+            $context = $adapter->getStreamContext();
+            $res = stream_context_set_option(
+                $context,
+                'ssl',
+                'verify_peer_name',
+                false
+            );
+            if (!$res) {
+                throw new \Exception('Unable to set sslverifypeername option');
+            }
+        } elseif ($adapter instanceof \Laminas\Http\Client\Adapter\Curl) {
+            $adapter->setCurlOption(CURLOPT_SSL_VERIFYHOST, false);
+        }
+
+        // Set timeout value
+        $timeout = 30;
+        $client->setOptions(
+            ['timeout' => $timeout, 'useragent' => 'VuFind', 'keepalive' => true]
+        );
+
+        // Set Accept header
+        $client->getRequest()->getHeaders()->addHeaderLine(
+            'Accept',
+            'application/json'
+        );
+
+        return $client;
+    }
+
+    /**
+     * Test that the url is realy working
+     *
+     * @param string $url image Url
+     *
+     * @return bool Http Client
+     */
+    protected function testUrlFunction($url)
+    {
+
+        // Use get_headers() function
+        $headers = @get_headers($url);
```

Comment:
It would probably be better to do this using the HTTP client object -- since get_headers is using a low-level PHP function, it means you're bypassing all the configuration encapsulated in the HTTP client. This could cause the function to break when running behind a proxy server (to offer just one example).

Reply:
I used get_headers because it is less load than geting the whole body. Is there such a function on the HTTP client that I missed?

Reply:
I think if you get the request object from the client, you can call `setMethod` on the request to do a HEAD request instead of GET or POST.

Reply:
It had actually, so forget my question.